array_analyzer/workflows/registration_workflow.py

```diff
@@ -96,20 +98,43 @@ def point_registration(input_dir, output_dir):
             stop_criteria=0.1,
             debug=constants.DEBUG,
         )
+
+        # Warn if fit is still bad
         if min_dist > constants.REG_DIST_THRESH:
-            warnings.warn("Registration failed, repeat with outlier removal")
+            spot_coords, im_filtered = img_processing.get_spot_coords(
+                image,
+                min_area=200,
+                min_thresh=100,
+                max_thresh=255,
+                min_circularity=0.1,
+                min_convexity=0.5,
+                min_dist_between_blobs=10,
+                min_repeatability=2,
+                sigma_gauss=10,
+                nbr_expected_spots=nbr_expected_spots,
+            )
             t_matrix, min_dist = registration.particle_filter(
                 fiducial_coords=fiducial_coords,
                 spot_coords=spot_coords,
                 particles=particles,
                 stds=np.array(constants.STDS),
                 stop_criteria=0.1,
-                remove_outlier=True,
                 debug=constants.DEBUG,
             )
-            # Warn if fit is still bad
             if min_dist > constants.REG_DIST_THRESH:
-                warnings.warn("Final registration failed, registration may be flawed")
+                warnings.warn("Registration failed, repeat with outlier removal")
```

Comment:
It seems like a lot to repeat the particle filter 3 times with different conditions... I'd rather get the spot detection right in the first pass. Perhaps we can leave it in for now with a TODO to improve spot detection?

Reply:
I agree. That would be ideal